main.go

```diff
@@ -342,3 +347,37 @@ func getUserClaims(i kratos_client.Identity, cr hydra_client.OAuth2ConsentReques
 
 	return ret
 }
+
+func setUpPrometheus() *prometheus.MetricsManager {
+	mm := prometheus.NewMetricsManagerWithPrefix("identity-platform-login-ui-operator", "http", "", "", "")
+	mm.RegisterRoutes(
+		"/api/kratos/self-service/login/browser",
+		"/api/kratos/self-service/login/flows",
+		"/api/kratos/self-service/login",
+		"/api/kratos/self-service/errors",
+		"/api/consent",
+		prometheus.PrometheusPath,
+	)
+
+	pages, err := ui.ReadDir("ui/dist/_next/static/chunks/pages")
```

Comment:
This is not correct, only `js` files are in that folder. You need to register the all the files with the `.html` suffix in `ui/dist`. 

I thought we should be able to reuse the object from https://github.com/canonical/identity-platform-login-ui/pull/32/files#diff-2873f79a86c0d8b3335cd7731b0ecf7dd4301eb19a82ef7a1cba7589b5252261R84 to do this

Reply:
Sorry, I somehow didn't notice the html pages in dist. I fixed that. So I have 2 reservations against using the middleware to "register" these endpoints:
- I want the package prometheus to do one thing well, which in this case is having a middleware to implement the instrumentation of http handlers. Having the middleware do scan the dir directory would adds complexity to an otherwise straightforward package. (I try to make packages as if they were separate modules with API to access their functionality. I think it makes the code base more manageable in the long run.)
- I think a middleware is well suited for work you need to do in the runtime every time you call the decorated function. In this case, scanning the ui/dist dir is something you only need to do at startup, because in our use case none of our pages should change in runtime.

Also I updated MetricsManager to store routes in a map instead of slice.